### DISCUSSION TITLE
[Feature Request] Maths Support

### DISCUSSION THREAD

**Discussion by garv-shah** [singerdmx/flutter-quill#904] [Feature Request]
I was wondering if it was possible to add maths support with LATEX? [This](https://pub.dev/packages/math_keyboard) plugin could be quite good to implement something like that.

I'd be happy to do it myself, but I'm not too sure how to add a new element to the page UI. I skimmed through the documentation but it wasn't quite clear, would I start with the `text_block.dart` file?

Thank you very much!! :)

**Comment by singerdmx** [comment]
Does quill define format for it?

**Comment by garv-shah** [comment]
format for the maths input?

**Comment by singerdmx** [comment]
Need the corresponding attribute for it. Please check quill documentation.

**Comment by garv-shah** [comment]
i presume it wouldn't be within the main quill package but rather in something likes mathquill

**Comment by singerdmx** [comment]
We need to make sure web version of react quill can be compatible

**Comment by garv-shah** [comment]
I'm pretty sure it is, since there are quite a few "react-mathquill" type packages which seem to work together from a quick google search

**Comment by singerdmx** [comment]
So what is it like for underlying json data?

**Comment by garv-shah** [comment]
I think it would just have a tag like normal, and then the math would be stored as LATEX

If you can guide me through how to add additional components, I could try to add it myself?

**Comment by singerdmx** [comment]
Please go through https://youtube.com/playlist?list=PLbhaS_83B97vONkOAWGJrSXWX58et9zZ2 if you don't understand what I mean by attribute.
Again, the first thing is to define json data as in

https://github.com/singerdmx/flutter-quill/blob/master/example/assets/sample_data.json

**Comment by garv-shah** [comment]
It's all maths stored as LATEX (or KATEX) inside the JSON
If you look at https://quilljs.com/docs/formats/ it's the "Formula" attribute, the same as normal maths in quilljs, and it can be edited with the package I linked above
I can't find an example of a JSON with the data, but if you can find a comprehensive quilljs one, you should be able to find it

**Comment by garv-shah** [comment]
oh, i watched the videos, and it seems you went through the same website I sent :P
so yeah, the attribute we're looking at is the embed that's not supported yet, which is formulas

**Comment by singerdmx** [comment]
Try https://codesandbox.io/s/3zf03 and you can intercept web traffic to figure out the json

**Comment by garv-shah** [comment]
i got it typed in, how do I get the JSON? sorry, i can't find it in dev tools

**Comment by singerdmx** [comment]
<img width="399" alt="image" src="https://user-images.githubusercontent.com/122956/182489027-84f9407e-df88-4efc-a026-c63dc230a1ab.png">

**Comment by garv-shah** [comment]
that just seems to be giving me html? is there a way to export the delta document?

**Comment by singerdmx** [comment]
It is possible if you know react

**Comment by garv-shah** [comment]
sorry you might have to do it for me :/

**Comment by garv-shah** [comment]
I made a commit to my fork [here](https://github.com/garv-shah/flutter-quill) and it seems easy enough to add a new toolbar entry and create a new widget to show the maths keyboard, buttt for some reason flutter quill has a very odd focus system which makes it so you can't type in the textbox like normal

Is there somewhere I can edit to fix up the focus, because it's currently quite broken

**Comment by singerdmx** [comment]
If you cannot figure out the json data, I am unsure what to talk about

**Comment by singerdmx** [comment]
This will be my final comment until json data is figure out. Please don't expect me to reply more

**Comment by garv-shah** [comment]
I'm not sure how to use react so I won't be able to figure out how to extract it from the link you sent

It seems like you've been able to type in expressions yourself so would you please be able to extract the JSON data for me?

**Comment by garv-shah** [comment]
ended up learning the basics of react to try and get the json data, here it is:

```json
{
  "ops": [
    {
      "insert": {
        "formula": "x=\\frac{-b\\pm\\sqrt{b^2-4ac}}{2a}"
      }
    }
  ]
}
```

would we be able to go ahead with implementing the formula attribute?

**Comment by singerdmx** [comment]
Sure. Please add this in https://github.com/singerdmx/flutter-quill/blob/master/example/assets/sample_data.json

**Comment by garv-shah** [comment]
https://github.com/singerdmx/flutter-quill/pull/899
done!
